src/QsCompiler/SyntaxProcessor/ExpressionVerification.fs

```diff
@@ -23,6 +25,28 @@ open Microsoft.Quantum.QsCompiler.Transformations.QsCodeOutput
 
 // utils for verifying types in expressions
 
+/// The context used for symbol resolution and type checking.
+type ResolutionContext<'a> =
+    { /// The symbol tracker for the parent callable.
+      Symbols : SymbolTracker<'a>
+      /// True if the parent callable for the current scope is an operation.
+      IsInOperation : bool
+      /// The return type of the parent callable for the current scope.
+      ReturnType : ResolvedType
+      /// The runtime capabilities for the compilation unit.
+      Capabilities : RuntimeCapabilities }
+
+/// <summary>Creates a resolution context for the specialization.</summary>
+/// <exception cref="Exception">Thrown if the specialization's parent does not exist.</exception>
```

Comment:
I would make this an ArgumentException rather than just a generic exception. 
```suggestion
/// <exception cref="ArgumentException">
/// Thrown if the given namespace manager does not contain all resolutions, 
/// or if the specialization's parent does not exist in the given namespace manager.
/// </exception>
```